tests/Unit/Attribute/Validator/PathValidatorTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace LastCall\DownloadsPlugin\Tests\Unit\Attribute\Validator;
+
+use LastCall\DownloadsPlugin\Attribute\Validator\PathValidator;
+use LastCall\DownloadsPlugin\Attribute\Validator\ValidatorInterface;
+use LastCall\DownloadsPlugin\Enum\Attribute;
+
+class PathValidatorTest extends AbstractValidatorTestCase
+{
+    private string $attribute = 'path';
+
+    public function testNull(): void
+    {
+        $this->parent->expects($this->once())->method('getName')->willReturn($this->parentName);
+        $this->attributeManager->expects($this->never())->method('get');
+        $this->expectUnexpectedValueException($this->attribute, 'is required');
+        $this->validator->validate(null);
+    }
+
+    public function getInvalidPathTests(): array
+    {
+        return [
+            [true, 'bool'],
+            [false, 'bool'],
+            [123, 'int'],
+            [12.3, 'float'],
+            [['key' => 'value'], 'array'],
+            [(object) ['key' => 'value'], 'stdClass'],
+        ];
+    }
+
+    /**
+     * @dataProvider getInvalidPathTests
+     */
+    public function testInvalidPath(mixed $invalidPath, string $type): void
+    {
+        $this->parent->expects($this->once())->method('getName')->willReturn($this->parentName);
+        $this->attributeManager->expects($this->never())->method('get');
+        $this->expectUnexpectedValueException($this->attribute, sprintf('must be string, "%s" given', $type));
+        $this->validator->validate($invalidPath);
+    }
+
+    public function getAbsolutePathTests(): array
+    {
+        return [
+            ['C:\Programs\PHP\php.ini'],
```

Comment:
this one leaves me with questions. We should not allow any path outside of web-root on any platform is a basic decompression rule, that when left out, leads to CVE's. Better to fail than allow these files to be able to be unpacked outside of a very specific directory structure.

What that base-path is, I am less unsure of, as all of these files should be within the non-public virtualhost specific directory of a build server. But I'm still keen on keeping folks build servers safe.

Reply:
Also please lets not have system or runtime config as examples.

Reply:
The base-path is the parent package's install path. Files are downloaded into this directory only.

Reply:
The example is replaced.